### DISCUSSION TITLE
requestSubscription withOffer alert message: Unable to Purchase

### DISCUSSION THREAD

**Discussion by caoyongfeng0214** [hyochan/react-native-iap#2629] [Q&A]
if not use withOffer, it work.

if used withOffer, the purchase process was unsuccessful.

On TestFlight, When the `requestSubscription` is called, the product information will pop up, and the user name and password will also be requested.

Alert message ' Unable to Purchase, Contact the developer for more infomation ' after submit password.

Here's my code：
```js
let productId = 'xxxx';
let appAccountToken = 'a-uuid';

// https://developer.apple.com/documentation/storekit/in-app_purchase/original_api_for_in-app_purchase/subscriptions_and_offers/generating_a_promotional_offer_signature_on_the_server?language=objc
let withOffer = await http.post('https://xxx/signature-with-offer', { ... });
/**
withOffer:
    {
        identifier:'vip_free_1_month',
        keyIdentifier:'jkjkjkjkjk',
        nonce:'a-uuid',
        signature:'abcdef',
        timestamp: 1232323232
    }
**/

requestSubscription({
    sku: productId,
    appAccountToken: appAccountToken,
    withOffer: withOffer
})
```

**Environment:**

- react-native-iap: 12.11.0
- react-native: 0.72.6
- Platforms (iOS):

Can anyone help?

**Comment by itsaliraxa** [comment]
@caoyongfeng0214 Did you get it fixed?
